subprojects/language-java/src/main/java/org/gradle/api/tasks/compile/CompileOptions.java

```diff
@@ -103,6 +103,8 @@ public CompileOptions(ObjectFactory objectFactory) {
         this.headerOutputDirectory = objectFactory.directoryProperty();
         this.release = objectFactory.property(Integer.class);
         this.incrementalAfterFailure = objectFactory.property(Boolean.class);
+        this.forkOptions = objectFactory.newInstance(ForkOptions.class);
+        this.debugOptions = new DebugOptions();
```

Comment:
Shouldn't `DebugOptions` be abstract, too?

Reply:
It can't be without rewriting some of the internal compiler classes to not use it directly.  This class gets serialized over to the compiler daemons.

Reply:
Should we still instantiate it via the `ObjectFactory`, just for consistency's sake?

Reply:
No, this can't be decorated at all or it fails serialization.

Reply:
In that case let's spell out the difference here in a source code comment, so the next person doesn't need to dig back to this PR comment.